kore/src/Kore/ModelChecker/Step.hs

```diff
@@ -97,6 +99,11 @@ data ProofState patt
     -- ^ State which can't be rewritten anymore.
   deriving (Show, Eq, Ord, Generic, Functor)
 
+-- TODO: is this right?
+instance TopBottom (ProofState patt) where
+    isTop _ = False
+    isBottom _ = False
+
```

Comment:
This is technically correct, but it's strange that we need it. I would prefer to get rid of whatever is using it. (I assume that somewhere we have a `MultiOr (ProofState _)`.)

Reply:
I think this could possibly turn into a larger change. It's needed [here](https://github.com/kframework/kore/blob/c28200738e0a5e8c2b9aaf6a6046b996f68082e3/kore/src/Kore/ModelChecker/Step.hs#L233), so we would probably have to change the type of `Results` or find a workaround.

Reply:
OK, let's leave it then.